src/Fmt_ast.ml

```diff
@@ -348,7 +357,20 @@ let fmt_constant c ~loc ?epi const =
           | Some s -> (s, `Preserve) )
         | _ -> (s, c.conf.escape_strings)
       in
+      let contains_pp_commands =
+        let is_substring substring = String.is_substring s ~substring in
+        List.exists ["@,"; "@;"; "@\\n"] ~f:is_substring
+      in
       match c.conf.break_string_literals with
+      | `Newlines when contains_pp_commands ->
+          let break_on_pp_commands in_ delim =
+            let pat = String.Search_pattern.create delim in
+            let with_ = delim ^ "\n" in
+            String.Search_pattern.replace_all pat ~in_ ~with_
```

Comment:
`String.substr_replace_all in_ ~pattern:delim ~with_` is shorter and equivalent since `pat` is used once and discarded.

Reply:
Done.